### PR TITLE
feat: phase 2, hexagonal site domain

### PR DESCRIPTION
**Pull request by ivanoats**
## Summary

Phase 2 of [the modernization plan](docs/plans/modernization.md):
introduce a hexagonal (ports & adapters) architecture for site data
without changing what the user sees.

- **`src/domain/`** — pure types: `Site`, `SiteId` (branded), `Coordinates`, `Facility` enum + `FacilitySet`. No framework deps.
- **`src/application/`** — `SiteRepository` port + `makeListSites` use-case factory.
- **`src/adapters/outbound/geojson-site-repository.ts`** — implements the port. Fetches `/data/ndwt.geojson` once, caches, de-dupes concurrent calls, tolerates the dirty source data (BOM-prefixed `web-scraper-order` keys, missing optional fields, `\"0\"`/`\"1\"` facility flags as strings).
- **`src/composition-root.ts`** — single repository instance wired to the use-case. UI imports only from here.
- **`src/components/map.tsx`** — refactored to consume `listSites()` and build OL `Feature`s from `Site[]`. Single source of truth; OL no longer fetches the GeoJSON URL itself. Cancellation guard on unmount during inflight load.

Dependency rule: `domain` → nothing. `application` → `domain`. `adapters` → `application` + `domain`. `ui` → `application` + `domain` (via composition root, never directly into adapters).

Also bumped the plan's tech-stack choice from Next.js 15 → 16 per the call we made today, and gitignored two more Claude harness runtime artifacts (`.agents/`, `skills-lock.json`).

## Test plan

- [x] \`npm run lint\`
- [x] \`npm run lint:md\`
- [x] \`npm run typecheck\`
- [x] \`npm test\` — 11 passing (3 new test files: \`FacilitySet\`, \`GeoJsonSiteRepository\` against the real GeoJSON, \`toSite\` mapper edge cases)
- [x] \`npm run build\`
- [x] Dev server starts and serves \`/data/ndwt.geojson\`
- [ ] **Browser smoke check (yours):** open \`localhost:5173\`, confirm all ~150 markers render exactly as before — I can't visually verify a rendered map from here, only that the data path types and unit tests pass.

## Bot review triage

After CI is green, sweep the inline comments and either fix, defer
(file follow-up), or dismiss with a one-line reason.

- [ ] Gemini Code Assist
- [ ] GitHub Copilot review
- [ ] DeepSource
- [ ] SonarCloud

## Notes for reviewer

- \`Bank\` and \`River\` value objects are **not** introduced this PR — the source data has irregular bank values like \`\"S- OR\"\` and \`\"South\"\` that don't fit a clean enum. Kept as raw strings on \`Site\`; will normalize in a later phase if the info panel needs it.
- The composition root is intentionally tiny and synchronous. When Phase 4 moves to Next.js, it'll grow a server-side variant (file-system loader for build time) alongside this one.

### CODE REVIEW

File: src/adapters/outbound/geojson-site-repository.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        throw new Error(
+          `Failed to fetch ${this.url}: ${res.status} ${res.statusText}`
+        );
+      }
+      const body = (await res.json()) as RawFeatureCollection;
+      const sites = body.features.map(toSite);
+      this.cache = sites;
+      this.inflight = null;
+      return sites;
+    })();
```
[the remark this answers is not in the document]
Comment (ivanoats) [reply src/adapters/outbound/geojson-site-repository.ts]: Real bug — fixed in d555233. The list() method now wraps the request in a `finally` that clears `this.inflight` whether the promise resolved or rejected, with an identity check (`this.inflight === request`) so concurrent calls don't accidentally clear each other. Also added a regression test (`retries after a failed fetch`) that asserts a 503 followed by a second list() call refetches and succeeds.

File: src/adapters/outbound/geojson-site-repository.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+export class GeoJsonSiteRepository implements SiteRepository {
+  private cache: readonly Site[] | null = null;
+  private inflight: Promise<readonly Site[]> | null = null;
+
+  constructor(private readonly url: string) {}
+
+  async list(): Promise<readonly Site[]> {
+    if (this.cache) return this.cache;
+    if (this.inflight) return this.inflight;
```
[the remark this answers is not in the document]
Comment (ivanoats) [reply src/adapters/outbound/geojson-site-repository.ts]: Tweaked in d555233 for clarity, but the original wasn't actually a bug — empty arrays are truthy in JS (`Boolean([]) === true`), so `if (this.cache)` would have correctly returned an empty cached result without refetching. Still, `!== null` is more intent-revealing, so applied to both `cache` and `inflight` checks.

File: src/adapters/outbound/geojson-site-repository.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      if (!res.ok) {
+        throw new Error(
+          `Failed to fetch ${this.url}: ${res.status} ${res.statusText}`
+        );
+      }
+      const body = (await res.json()) as RawFeatureCollection;
+      const sites = body.features.map(toSite);
+      this.cache = sites;
+      this.inflight = null;
+      return sites;
```
[the remark this answers is not in the document]
Comment (ivanoats) [reply src/adapters/outbound/geojson-site-repository.ts]: Same root cause as Gemini's HIGH comment — fixed in d555233. Used `request.finally(...)` instead of the `try/finally` shape you suggested (functionally equivalent, slightly less indentation). Includes a regression test covering 503-then-success.

File: src/components/map.tsx [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
 import TileLayer from 'ol/layer/Tile';
 import VectorLayer from 'ol/layer/Vector';
 import { fromLonLat } from 'ol/proj';
 import OSM from 'ol/source/OSM';
 import VectorSource from 'ol/source/Vector';
 import { useEffect, useRef } from 'react';
 
+import { listSites } from '../composition-root';
+import type { Site } from '../domain';
+
```
[the remark this answers is not in the document]
Comment (ivanoats) [reply src/components/map.tsx:12]: The PR description was imprecise, sorry. The actual hex-arch dependency rule (in docs/plans/modernization.md) is: `ui → application + domain` (never `adapters`). Importing pure domain types like `Site` is allowed and idiomatic — the domain is the shared vocabulary across all layers. What's prohibited is the UI reaching into `src/adapters/` directly, which it doesn't. Composition root exists for runtime *wiring* (the bound use-case), not as a re-export hub for types. Dismissing.
